File: Data Structures & Algorithms/sort-colors/submission-2.go
func sortColors(nums []int) {
	l := 0
	r := len(nums) - 1
	i := 0

	for i <= r{
		if nums[i] == 0{
			nums[i] , nums[l] = nums[l], nums[i]
			l += 1
			i += 1
		}else if nums[i] == 2{
			nums[i] , nums[r] = nums[r], nums[i]
			r -= 1
		}else{
			i += 1
		}
	}

}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-2.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        dp = {}

        def rec(i,s1,s2):
            if i == len(nums) and s1 == s2:
                return True
            if i >= len(nums):
                return False
            if dp.get((i,s1,s2)) is not None:
                return dp[(i,s1,s2)]
            cur = nums[i]

            a = rec(i+1,s1+cur,s2)
            b = rec(i+1,s1,s2+cur)
            dp[(i,s1,s2)] = a or b
            return a or b

        return rec(0,0,0)

        